### DISCUSSION TITLE
Graphcache: Resolving partial data from the same entity

### DISCUSSION THREAD

**Discussion by cvpcasada** [urql-graphql/urql#1747] [Q&A] [answered]
Hello, I just started reading up on how normalized caching works so I don't fully understood what it do. 
Right now, I have two queries that resolves to different fields but from the same entity `File`

**Query 1: Returns file fields**

```gql
query File($id: ID!) {
  file: File(id: $id) {
    id
    fileName
  }
}
```

**Query 2: Returns file comment field, comments is a connection**
```gql
query FileComments($id: ID!) {
  file: File(id: $id) {
    id
    comments { # type is FileCommentConnection
      edges {
        node {
          id
          text
        }
      }
    }
  }
}
```

My question is on how to I resolve this so that the cache will merge the results from the queries to the same entity. Btw, I have introspection schema data enabled on the cacheExchange.

**Comment by JoviDeCroock** [comment, marked as answer]
Hey,

Yes the cache will merge these two results so imagine the following scenario:

```
{
  id:1
  __typename: "File",
  comments: {
    edges: [{ node: { id: 1, __typename: 'Comment' } }]
  }
}
```

Then the cache will save the records `File:1.id=1`, ... and save that File:1 links to the comments it has received, this is how normalized caching will handle this.

Now when `comments` is marked as optional and you've done the first query you'll get a `partial` result (recognizable by stale: true). This means that we'll give you the data from the first query while we resolve the additional fields (comments) from the second.

**Reply by cvpcasada** [reply]
Thanks for the answer, made it a bit more clearer to me now! 😁

**Reply by thenbe** [reply]
> Now when `comments` is marked as optional

@JoviDeCroock Hi. I'm new to all of this, but where/how do you mark `comments` as optional?

**Reply by JoviDeCroock** [reply]
This is a server-side semantic, we allow you to pass in an introSpectionResult through [schema awareness](https://formidable.com/open-source/urql/docs/graphcache/schema-awareness/)

**Reply by thenbe** [reply]
Thanks for the quick response! How can I know if I have successfully passed in my introspectionResult? 

Every time I refresh my site, the urql client sends a network request to my graphql server to ask for an introspectionResult. How can I tell the client to use the local introspectionResult that I passed in instead of contacting my graphql server every time?

**Reply by JoviDeCroock** [reply]
That's probably coming from urql-devtools 😅 graphcache won't send one

**Reply by thenbe** [reply]
I finally got it to work by passing in this to the cacheExchange

```
  resolvers: {
    Query: {
      spoon: (_, args) => ({ __typename: 'Spoon', id: args.id }),
    },
  },
```
For some reason, I assumed that urql would do this automatically for me once I gave it the schema Introspection Result (probably because it does everything else automatically 😅) . Just to be sure though, I need to do this manually for all my types? AKA 

```
  resolvers: {
    Query: {
      spoon: (_, args) => ({ __typename: 'Spoon', id: args.id }),
      fork: (_, args) => ({ __typename: 'Fork', id: args.id }),
      chair: (_, args) => ({ __typename: 'Chair', id: args.id }),
      // etc etc etc
    },
  },
```

**Reply by JoviDeCroock** [reply]
Yes, we can't assume that you don't want to call the server so we don't make those connections automatically
